homeassistant/components/sensor/teamspeak.py

```diff
@@ -0,0 +1,112 @@
+"""
+homeassistant.components.sensor.teamspeak
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Monitors a teamspeak server
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.teamspeak/
+"""
+import logging
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import convert
+
+REQUIREMENTS = ['ts3==0.7.1']
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Teamspeak Server'
+ICON = 'mdi:microphone'
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Teamspeak sensor. """
+    name = config.get('name', DEFAULT_NAME)
+    host = config.get('host')
+    username = config.get('username')
+    password = config.get('password')
+    virtualserver_id = convert(config.get('virtualserver_id'), int, 0)
+
+    if host is None:
+        _LOGGER.error('No teamspeak host specified')
+        return
+    elif username is None:
+        _LOGGER.error('No teamspeak username specified')
+        return
+    elif password is None:
+        _LOGGER.error('No teamspeak password specified')
+        return
+
+    add_devices([
+        TeamspeakSensor(name, host, username, password, virtualserver_id)
+    ])
+
+
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+class TeamspeakSensor(Entity):
+    """ A Teamspeak Server sensor. """
+
+    def __init__(self, name, host, username, password, virtual_server_id):
+        self._name = name
+        self._state = None
+        self._unit_of_measurement = 'Users'
+        self._ts3conn = None
+
+        self._host = host
+        self._username = username
+        self._password = password
+        self._virtual_server_id = virtual_server_id
+
+        self.connect()
+        self.update()
+
+    @property
+    def name(self):
+        """ The name of the sensor. """
+        return self._name
```

Comment:
Thank you for the suggestion; I think the user should have the option to set it in config, but if name isn't set it defaults to the TS server name. For example, the server I'm monitoring has a long name and this looks bad in Home Assistant.

So, I added a check in init (this seems best because name should only be set at initialization and doesn't change?) to see if config name is none
However, I'm a bit stuck with how exactly to write this line:
`self._name = self._ts3conn.serverlist()[self._virtual_server_id]['virtualserver_name']`
is too long for lynting, so how do I split this?

1)Seems ridiculous?
```
        self._name = self._ts3conn.serverlist()[
            self._virtual_server_id
            ][
                'virtualserver_name'
            ]
```
2)This also works, but is using many variables really better than just a long line?
```
            serverlist = self._ts3conn.serverlist()
            virtualserver = serverlist[self._virtual_server_id]
            self._name = virtualserver['virtualserver_name']
```

Thank you